src/sds_data_model/dataframe.py

```diff
@@ -0,0 +1,254 @@
+from logging import getLogger, INFO, StreamHandler, Formatter
+from typing import Any, Dict, Optional,  Type, TypeVar
+from typing_extensions import Self
+from pathlib import Path
+from inspect import ismethod, signature
+
+from dataclasses import dataclass
+from typing_extensions import Self
+from pyspark.pandas import read_excel
+from pyspark.sql import DataFrame as SparkDataFrame
+
+from pyspark.pandas import DataFrame, read_excel
```

Comment:
Do you still need to import the Pandas on Spark DataFrame? If so, alias as follows to avoid name collision with the actual `pandas DataFrame`:  

```suggestion
from pyspark.pandas import DataFrame as SparkPandasDataFrame
from pyspark.pandas import read_excel
```